OpenRA.Mods.Common/Activities/Transform.cs

```diff
@@ -28,61 +28,94 @@ public class Transform : Activity
 		public bool SkipMakeAnims = false;
 		public string Faction = null;
 
+		readonly bool orderedMove;
+		readonly Target target = Target.Invalid;
+		readonly CPos cell;
+		IMove move;
+		bool hasMoved;
+
 		public Transform(Actor self, string toActor)
 		{
 			ToActor = toActor;
 		}
 
+		public Transform(Actor self, Target target)
+		{
+			this.target = target;
+			orderedMove = target.Type == TargetType.Terrain || (target.Type == TargetType.Actor && target.Actor != self);
+			if (orderedMove)
+				cell = self.World.Map.Clamp(self.World.Map.CellContaining(target.CenterPosition));
+		}
+
 		protected override void OnFirstRun(Actor self)
 		{
-			if (self.Info.HasTraitInfo<IFacingInfo>())
-				QueueChild(self, new Turn(self, Facing));
+			if (!orderedMove)
+			{
+				if (self.Info.HasTraitInfo<IFacingInfo>())
+					QueueChild(self, new Turn(self, Facing), true);
 
-			if (self.Info.HasTraitInfo<AircraftInfo>())
-				QueueChild(self, new HeliLand(self, true));
+				if (self.Info.HasTraitInfo<AircraftInfo>())
+					QueueChild(self, new HeliLand(self, true), true);
+			}
 		}
 
 		public override Activity Tick(Actor self)
 		{
-			if (IsCanceling)
-				return NextActivity;
-
 			if (ChildActivity != null)
 			{
 				ActivityUtils.RunActivity(self, ChildActivity);
 				return this;
 			}
 
-			// Prevent deployment in bogus locations
-			var transforms = self.TraitOrDefault<Transforms>();
-			if (transforms != null && !transforms.CanDeploy())
-			{
-				Cancel(self, true);
+			if (IsCanceling)
 				return NextActivity;
-			}
 
-			foreach (var nt in self.TraitsImplementing<INotifyTransform>())
-				nt.BeforeTransform(self);
-
-			var makeAnimation = self.TraitOrDefault<WithMakeAnimation>();
-			if (!SkipMakeAnims && makeAnimation != null)
+			if (!hasMoved)
 			{
-				// Once the make animation starts the activity must not be stopped anymore.
-				IsInterruptible = false;
+				move = self.TraitOrDefault<IMove>();
+				if (orderedMove && move != null)
+				{
+					hasMoved = true;
+					QueueChild(self, move.MoveTo(cell, 8), true);
+					return this;
+				}
```

Comment:
Same as the previous comment: this is just wrapping around a regular move order for undeployed actors, right?

Reply:
Yes, see above.